Framework/Core/include/Framework/AnalysisDataModel.h

```diff
@@ -24,9 +24,10 @@ DECLARE_SOA_STORE();
 
 namespace collision
 {
-// DECLARE_SOA_COLUMN(TimeframeId, timeframeId, uint64_t);
 DECLARE_SOA_COLUMN(RunNumber, runNumber, int);
+// TODO shall globalBC and globalBCUncertainty  be stored in the same 64bit? Is this bad for a column we use for indexing?
```

Comment:
Please comment

Reply:
GlobalBC is supposed to be a column in a different table, is it not? And Collisions is supposed to have an index to that table, pointing to row numbers.

Reply:
I am not sure about int-like GlobalBCUncertainty. Do we really need it? As far as I understand, this uncertainty is already encoded in the float-like collision time uncertainty. I would simply omit this field.

Reply:
@aalkin No, it is not a direct reference, we don't use row numbers. The GlobalBC is a guess, and there may not be even an entry with that specific GlobalBC in the Trigger table (for pp)
@ekryshen @shahor02 had asked during the meeting for a field encoding the uncertainty we have on that value, and if that uncertainty is a strict range or Gaussian

Reply:
@jgrosseo then I do not understand how this data model will be connected. Collision table will not have any indices to anything in this form. The idea, as far as I remember, was to have a dedicated table with BCs, and then both triggers and collisions tables refer to it. This way there is a clear way to access everything and relatively easy to access other compatible BCs for a collisions.

Reply:
The table with BCs exist, it is called Triggers, but up to now it is not used for the association, but for the trigger info.
Just from the logic (then we can make an intermediate on-the-fly table if that helps), it would work like this (pseudo-code):
if (collision.globalBCUncertainty() == 0) {
  auto entry = Calos.getEntryWithGlobalBC(collision.globalBC())
  // do something
} else {
  for (auto i = collision.globalBC() - collision.globalBCUncertainty(); i<=collision.globalBC() + collision.globalBCUncertainty(); i++) {
  auto entry = Calos.getEntryWithGlobalBC(i)
  // do something
  }
}

Reply:
@jgrosseo then you will have column with globalBC in collisions and column with all globalBCs in a different table, and data model tables with not have an index to neither of the two. This goes against how the rest of the model is designed and prevent us from using (and optimizing) indices and accesses and introduces manual loops. 

Compare your code to this hypothetical example:
`if (collision.BCs().szie() > 1) {
  auto calos = GroupBy<Calos, GlobalBC>(collision.BCs());
  for (auto& entry: calos) {
    // do something
  }
}`

Reply:
Concerning the ``GlobalBCUncertainty``: don't remember what exactly was said at the meeting, but my point was that the uncertainty assigned to the collision time can be either gaussian (e.g. from the fit of the vertex with global tracks time info) or a range (e.g. defined by the ITS readout window). As far as I understand, this is encoded in ``CollisionTime`` and ``CollisionTimeRes`` (is there an flag telling the nature of the uncertainty? I guess in any case the ``collision`` should have a flags field). As for the GlobalBC and its uncertainty, how is it supposed to be built? Just by converting CollisionTime to BC? Combining that with FIT info?

Reply:
If the plan is to do loops around GlobalBC in case of significant event time uncertainties, it might be better to create dynamic columns like GlobalBCMin and GlobalBCMax and fill them according to the contents of CollisionTimeRes.

Reply:
So what is the minimal information that would allow us to calculate and use additional BCs? Do we need the full uncertainty for analysis? Or is it enough to have a list of BCs (maybe with probabilities attached)?

Reply:
@aalkin : should depend on the error on the interaction time. E.g. if you have vertex from ITS tracks only, then all BCs in the ITS strobe (a few \mus) are potential candidates (+ other info can used, e.g. FIT). If there is a low-mult vertex fitted with global tracks, you will have a gaussian time uncertainty of ~100ns, then one should define a confidence limit where the BC will be selected. 
All these is in principle, whether in practice we will do this, I don't know. I would say such details are a fine-tuning, I would leave room for them but would not try to finalize now.

Reply:
@shahor02 thank you for the explanation. In this case the straightforward way is to have a single column table with list of all global BCs, then everything else, including `Triggers` table has an index to it. Then, either in `Collisions` table directly, or, for example, in `CollisionsExtra` we can have whatever we need to calculate additional BCs for a given collision with a helper class. In addition we will need some grouping tools and on-the-fly index generation in the Framework to have more cross-table access.

Reply:
Thanks! 
1) @shahor02 So I remove GlobalBCUncertainty, and we use the CollisionTimeRes for it. Here one bit should be reserved for the nature of the uncertainty. A dynamic column should extract this value. I don't know yet how we would calculate the GlobalBC. I'd assume this happens in the reconstruction?
2) @ekryshen If the uncertainty is Gaussian, can we clearly define min/max. Wouldn't it depend on the analysis? If one can cut tighter or not?
3) @aalkin So in any case, we need an intermediate table to which one could then index. Either this one is persistent or built on the fly. What is confusing for me: if globalBC is unique, why do I need an additional table which links an index with globalBC. Cannot globalBC be the primary key?

Reply:
@jgrosseo indexed access is based on row number and is therefore very fast (because it is direct, there are no checks), access by value lookup (which we will implement anyway) would be considerably slower. Especially if it needs to be done many times, in a loop for example. 

The problem here is the cross-access, for example we want calo info for a collision, we can go to calo table and start looking for a row with the same value in BCid as for the collision we are inspecting. This, of course, is not optimal. However, we can on-the-fly extend the BC table with indices to every other (requested) table, and then in the user code things will be as simple as `collision.BC().calo().somecolumn()`, etc.

Reply:
@shahor02 How do we deal with the assignment of the muon tracks? 
With a MID hit, we have the exact GlobalBC. If we have tracks without we would need an uncertainty column also there

Reply:
Not sure muon tracks w/o MID are allowed, better @aphecetche comment.

Reply:
Well, I'd let @mwinn2 comment on whether or not in the end we'll use unmatched (with MID) muon tracks at all, as there are some PWG-driven discussions (in which I think @ekryshen is involved as well) about the physics case(s) behind, that I must confess I'm not very closely following.

Reply:
Regarding muon tracks without MID hit. As far as I know, they are not kept in the current framework. However we would definitely benefit if we could store and analyse them. They would be useful in several cases:
1) Data driven estimate of MID matching efficiency.
2) Significantly higher efficiency for dimuons, especially at low masses. Especially for UPC.

We started this discussion with muon experts a month ago. We should probably resurrect it. But github is not the best place to discuss it.

Reply:
@jgrosseo Regarding your question: "If the uncertainty is Gaussian, can we clearly define min/max. Wouldn't it depend on the analysis? If one can cut tighter or not?"
I agree. Anyway, my point is to make a dynamic column(s) for the bc uncertainty (or bc range) linked to CollisionTimeRes. I think we need a use case to decide how to proceed. For the moment we can simply omit it.

Reply:
@ekryshen 
1) For now I assume we have an MID hit, but have added a NOTE in the code. Please follow the discussion outside of this PR, and then update us
2) After discussion with @aalkin and @ktf, we propose to have the Collision always point to the closest trigger in the Trigger table. In addition, we store a field "estimatedGlobalBCOffset" which gives the offset between the trigger we point to, i.e. collision.trigger().globalBC(), and the one estimated from the collision time

Reply:
After discussion with @ekryshen , we decided to store all BC in the Trigger table, so there is no need any more for the estimatedGlobalBCOffset column